lib/Firtool/Firtool.cpp

```diff
@@ -243,3 +258,58 @@ LogicalResult firtool::populateHWToSV(mlir::PassManager &pm,
 
   return success();
 }
+
+LogicalResult
+firtool::populatePrepareForExportVerilog(mlir::PassManager &pm,
+                                         const FirtoolOptions &opt) {
+
+  // Legalize unsupported operations within the modules.
+  pm.nest<hw::HWModuleOp>().addPass(sv::createHWLegalizeModulesPass());
+
+  // Tidy up the IR to improve verilog emission quality.
+  if (!opt.disableOptimization)
+    pm.nest<hw::HWModuleOp>().addPass(sv::createPrettifyVerilogPass());
+
+  if (opt.stripFirDebugInfo)
+    pm.addPass(circt::createStripDebugInfoWithPredPass([](mlir::Location loc) {
+      if (auto fileLoc = loc.dyn_cast<FileLineColLoc>())
+        return fileLoc.getFilename().getValue().endswith(".fir");
+      return false;
+    }));
+
+  if (opt.stripDebugInfo)
+    pm.addPass(circt::createStripDebugInfoWithPredPass(
+        [](mlir::Location loc) { return true; }));
+
+  // Emit module and testbench hierarchy JSON files.
+  if (opt.exportModuleHierarchy)
+    pm.addPass(sv::createHWExportModuleHierarchyPass());
```

Comment:
Before, when this was called in firtool.cpp, the `outputFilename` was passed as an argument. I think this still needs to be passed to maintain whatever the previous behavior was.